pkg/datarepair/checker/checker_test.go

```diff
@@ -104,3 +105,77 @@ func TestOfflineNodes(t *testing.T) {
 		assert.Equal(t, expectedOffline, offline)
 	})
 }
+
+func TestIdentifyIrreparableSegments(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 3, UplinkCount: 0,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		time.Sleep(2 * time.Second)
+
+		const numberOfNodes = 10
+		pieces := make([]*pb.RemotePiece, 0, numberOfNodes)
+		// use online nodes
+		for i, storagenode := range planet.StorageNodes {
+			pieces = append(pieces, &pb.RemotePiece{
+				PieceNum: int32(i),
+				NodeId:   storagenode.ID(),
+			})
+		}
+
+		// simulate offline nodes
+		expectedLostPieces := make(map[int32]bool)
+		for i := len(pieces); i < numberOfNodes; i++ {
+			pieces = append(pieces, &pb.RemotePiece{
+				PieceNum: int32(i),
+				NodeId:   storj.NodeID{byte(i)},
+			})
+			expectedLostPieces[int32(i)] = true
+		}
+		pointer := &pb.Pointer{
+			Remote: &pb.RemoteSegment{
+				Redundancy: &pb.RedundancyScheme{
+					MinReq:          int32(4),
+					RepairThreshold: int32(8),
+				},
+				PieceId:      "fake-piece-id",
+				RemotePieces: pieces,
+			},
+		}
+
+		// put test pointer to db
+		pointerdb := planet.Satellites[0].Metainfo.Service
+		err := pointerdb.Put(pointer.Remote.PieceId, pointer)
+		assert.NoError(t, err)
+
+		checker := planet.Satellites[0].Repair.Checker
+		err = checker.IdentifyInjuredSegments(ctx)
```

Comment:
Note directly invoking those service methods may cause races in the future.

Reply:
Yes, I know but I think we don't have easy way to disable services triggered by timers. I will add comment about it.